src/main.rs

```diff
@@ -340,8 +341,12 @@ fn create_main(xi_peer: XiPeer) -> Result<(HWND, Rc<Box<WndProc>>), Error> {
         let main_state = MainWinState::new();
         let main_win: Rc<Box<WndProc>> = Rc::new(Box::new(
             MainWin::new(xi_peer, main_state)));
-        let width = 500;  // TODO: scale by dpi
-        let height = 400;
+
+        // Simple scaling based on System Dpi (96 is equivalent to 100%)
+        let dpi = GetDpiForSystem() as i32;
+        let width = 500 * (dpi/96);
```

Comment:
I think this code will only work well for dpi values that are an integer multiple of 96.

Other than that, the approach is fine for now. I've experimented some with per-monitor dpi, but have come to the conclusion that (a) it really needs a manifest to work properly and (b) it only works on Creators or higher, and earlier versions should stick with System to avoid all kinds of odd behavior.